From 513fdc85ebca34a0df1b0e2cbc3c6cac4682e3c0 Mon Sep 17 00:00:00 2001
From: Raphael Gallais-Pou <raphael.gallais-pou@foss.st.com>
Date: Thu, 6 Apr 2023 16:51:30 +0200
Subject: [PATCH 505/529] ARM: dts: stm32: define a 128MB CMA area on
 stm32mp157a-ev1-scmi

Grow the CMA size to 128MB. The Continuous Memory Allocator allows to
create large chunks of continuous physical memory addresses. This could
be typically the reserved memory for the GPU.

Change-Id: Ia617145563d876bc3535df7eae72fe9d20bf2a5a
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/linux-stm32/+/299472
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
Tested-by: Raphael GALLAIS-POU <raphael.gallais-pou@foss.st.com>
Reviewed-by: Raphael GALLAIS-POU <raphael.gallais-pou@foss.st.com>
Reviewed-by: Philippe CORNU <philippe.cornu@foss.st.com>
Reviewed-by: Yannick FERTRE <yannick.fertre@foss.st.com>
Domain-Review: Philippe CORNU <philippe.cornu@foss.st.com>
---
 arch/arm/boot/dts/stm32mp157a-ev1-scmi.dtsi | 9 +++++++++
 1 file changed, 9 insertions(+)

--- a/arch/arm/boot/dts/stm32mp157a-ev1-scmi.dtsi
+++ b/arch/arm/boot/dts/stm32mp157a-ev1-scmi.dtsi
@@ -10,6 +10,15 @@
 			reg = <0xfe000000 0x2000000>;
 			no-map;
 		};
+
+		/* global autoconfigured region for contiguous allocations */
+		linux,cma {
+			compatible = "shared-dma-pool";
+			reusable;
+			size = <0x8000000>;
+			alignment = <0x2000>;
+			linux,cma-default;
+		};
 	};
 };
 
